packages/styles/src/components/button.scss

```diff
@@ -95,84 +79,70 @@
   }
 }
 
-// Primary
-.btn-primary {
-  border-color: transparent;
-  background-color: var(--post-gray-80);
-  color: var(--post-contrast-color-inverted);
+// Type variants
+@each $variant in button.$btn-variants {
+  .btn-#{$variant} {
+    $color: if($variant == 'default', 'secondary', if($variant == 'link', 'tertiary', $variant));
 
-  &:disabled {
-    background-color: color.$gray-10;
-    border-color: color.$gray-60;
-  }
-
-  .pi {
-    filter: invert(1);
-    forced-color-adjust: none;
-  }
+    @if ($variant != 'link' and $variant != 'tertiary') {
+      box-shadow: tokens.get('button-elevation');
+    }
 
-  @include color-mx.on-dark-background {
-    background-color: var(--post-contrast-color);
-    color: var(--post-gray-10); // Apply same color as secondary/tertiary button on dark theme
+    @include button-mx.button-variant-def('enabled', $color);
 
     &:disabled {
-      color: var(--post-contrast-color);
-      background-color: color.$white-alpha-20;
-      border-color: var(--post-contrast-color);
+      @include button-mx.button-variant-def('disabled', $color);
     }
-  }
 
-  @include color-mx.on-dark-background() {
-    .pi {
-      filter: invert(0);
+    @include utilities.not-disabled-hover() {
+      @include button-mx.button-variant-def('hover', $color);
+    }
+
+    @include color-mx.on-dark-background {
```

Comment:
This old mixin should not really be used anymore, at least not where we have tokens. For example for the border, we just need to define

```CSS
button {
  border-color: tokens.get('button-primary-enabled-stroke');
}
```
 and the dark mode tokens will take care of mapping to the correct color in each mode.